bin/salt-registerd

```diff
@@ -183,93 +223,276 @@ class Saltwalk:
+    """
+    sid = is_minion_registered(minion)
+    if not sid:
+        logger.warning("%s is not registered. Interfaces ignored.", minion)
+        return
+
+    profile = dict()
+    interfaces = event['data']['return']
+    logger.info('Got network interfaces for %s', sid)
+    logger.debug(pp.pprint(interfaces))
+
+    profile['class'] = 'NETINTERFACES'
+    for iface, details in interfaces.iteritems():
+        profile[iface] = dict()
+        profile[iface]['ipv6'] = list()
+        profile[iface]['hwaddr'] = details['hwaddr']
+        # FIXME: how to get the iface module with Salt?
+        profile[iface]['module'] = 'Unknown'
+
+        # only one ipv4 addr supported
+        for ipv4net in details['inet']:
+            profile[iface]['ipaddr'] = ipv4net['address']
+            profile[iface]['netmask'] = ipv4net['netmask']
+            profile[iface]['broadcast'] = ipv4net['broadcast']
+            break
+
+        for ipv6net in details['inet6']:
+            ipv6net['scope'] = 'Unknown'
+            ipv6net['addr'] = ipv6net['address']
+            ipv6net['netmask'] = ipv6net['prefixlen']
+            profile[iface]['ipv6'].append(ipv6net)
 
         server = rhnServer.search(int(sid))
+
         # No need to delete the hardware as the class seems to ovewrite
         # the previous value
         server.add_hardware(profile)
         server.save_hardware()
 
-    def _process_package_list_result(self, minion, ev):
-        logging.info('Got package list for %s' % minion)
-        if not ev['data']['success']:
-            return
-
-        package_list = ev['data']['return']
-        # Use xml-rpc for now to upload the packages
-        sid = self._is_registered(minion)
-        logging.info('Updating package list for Spacewalk sid=%s' % sid)
-        if sid:
-            packages = [{'name': 'zlib', 'epoch': '', 'version': '1.2.7', 'release': '0.10.128', 'arch': 'x86_64', 'installtime': 1413297811}]
-            server = rhnServer.search(int(sid))
-            server.dispose_packages()
-            for package in packages:
-                server.add_package(package)
-            server.save_packages()
-
-kernel = Saltwalk()
-kernel.run()
+
+def format_package_list(packages):
+    """
+    Take the package list provided by salt and
+    format it as spacewalk expects it.
+    """
+    # FIXME: Fake data needs to be sourced correctly
+    def frmt_pkg(name, version):
+        return {'name': name, 'version': version,
+                'epoch': '', 'release': 'unknown',
+                'arch': 'x86_64', 'installtime': 1413297811}
+    return [frmt_pkg(name, version) for name, version in packages.items()]
+
+
+def process_package_list_result(minion, event):
+    """
+    Updates the server registration of the minion with
+    the list of packages reported by salt.
+    """
+    logger.info('Got package list for %s', minion)
+
+    sid = is_minion_registered(minion)
+
+    package_list = format_package_list(event['data']['return'])
```

Comment:
Maybe this makes sense to be executed only if `sid` is around? I would move this below, like:
```python
for package in format_package_list(event.get('data', {}).get('return', {})):
    server.add_package(.....
```